integration/sync/test_mongodbQueue.js

```diff
@@ -0,0 +1,86 @@
+var MongodbQueue = require('../../lib/sync/mongodbQueue');
+var metrics = require('../../lib/sync/sync-metrics').init();
+var async = require('async');
+var assert = require('assert');
+var helper = require('./helper');
+
+var queueName = 'test_mongodb_queue';
+var mongoDBUrl = 'mongodb://127.0.0.1:27017/' + queueName;
+
+var mongodb;
+var queue;
+module.exports = {
+  'test mongodb queue': {
+    'before': function(done) {
+      helper.dropCollection(mongoDBUrl, queueName, function(err, db){
+        if (err) {
+          return done(err);
+        }
+        mongodb = db;
+        return done();
+      });
+    },
+
+    'after': function(done) {
+      if (queue) {
+        queue.stopPruneJob();
+      }
+      done();
+    },
+
+    'test mongo queue add, get, search and prune': function(done) {
```

Comment:
@wei-lee I'm not sure. It's possible that instead of it being a problem with this test, that it's a problem with how `npm test` runs the tests: it could be that the pre-test script that configures the container with MongoDB doesn't wait for it to be ready, and then the test can't connect properly, or something?